Engine/ScriptAnalyzer.cs

```diff
@@ -875,39 +875,41 @@ private List<string> GetValidCustomRulePaths(string[] customizedRulePath, PathIn
 
         private IEnumerable<T> GetInterfaceImplementationsFromAssembly<T>(string ruleDllPath) where T : class
         {
-            var fileName = Path.GetFileNameWithoutExtension(ruleDllPath);
+            string fileName = Path.GetFileNameWithoutExtension(ruleDllPath);
             var assemblyName = new AssemblyName(fileName);
             outputWriter.WriteVerbose(string.Format("Loading Assembly:{0}", assemblyName.FullName));
 
-            var dll = Assembly.Load(assemblyName);
-            var rules = new List<T>();
+            Assembly dll = Assembly.Load(assemblyName);
+
             if (dll == null)
             {
                 outputWriter.WriteVerbose(string.Format("Cannot load {0}", ruleDllPath));
-                return rules;
+                return Enumerable.Empty<T>();
             }
-            foreach (var type in dll.ExportedTypes)
+
+            var rules = new List<T>();
+            foreach (Type type in dll.ExportedTypes)
             {
-                var typeInfo = type.GetTypeInfo();
-                if (!typeInfo.IsInterface
-                    && !typeInfo.IsAbstract
-                    && typeInfo.ImplementedInterfaces.Contains(typeof(T)))
+                if (type.IsInterface
+                    || type.IsAbstract
+                    || !typeof(T).IsAssignableFrom(type))
+                {
+                    continue;
+                }
+
+                outputWriter.WriteVerbose(
```

Comment:
This verbose log makes execution quite noisy without adding much value unless I am missing something. It seems this was maybe useful for you for development but I think it would make sense to take this out at the end.

Reply:
Hmmm, I don't think I added this. But happy to turn it off. There are a couple of places where I think maybe code got in from a rebase?